master/docs/manual/configuration/global.rst

```diff
@@ -220,6 +222,16 @@ This URL must end with a slash (``/``).
 
 When status notices are sent to users (e.g., by email or over IRC), :bb:cfg:`buildbotURL` will be used to create a URL to the specific build or problem that they are being notified about.
 
+Startup
+~~~~~~~
+
+The master instance is launched by a script that verifies that master has been launched successfully by following the logs.
+The script will raise an error if it does not detect any progress in 10 seconds by default.
+On very large installations the startup may be very slow, so the timeout is configurable by the ``start_timeout`` key in the config.
+
+::
+    c['start_timeout'] = 30
```

Comment:
I believe from the code that this start_timeout is taken from the command line arguments, not from the config file.

It looks like it is not properly declared in the runner.py file

https://github.com/buildbot/buildbot/blob/9f5d9de5bd8ed0815c574e10526cff2e5aaef34a/master/buildbot/scripts/runner.py#L199

Reply:
Indeed I assumed wrong things from the initial PR and assumed unit tests will catch everything. Fixed now: removed bad documentation, added extra options and tests for the start script itself.